v3/src/components/web-view/web-view-url-modal.tsx

```diff
@@ -0,0 +1,94 @@
+import {
+  Button, FormControl, Input, ModalBody, ModalFooter, ModalHeader, Tooltip
+} from "@chakra-ui/react"
+import React, { useRef, useState } from "react"
+import { CodapModal } from "../codap-modal"
+import t from "../../utilities/translation/translate"
+
+interface IProps {
+  currentValue?: string
+  isOpen: boolean
+  onAccept: (value: string) => void
+  onClose: () => void
+}
+
+export const WebViewUrlModal = ({ currentValue="", isOpen, onAccept, onClose }: IProps) => {
+  const inputRef = useRef<HTMLInputElement | null>(null)
+  const [value, setValue] = useState(currentValue)
+
+  const applyValue = () => {
+    onAccept(value)
+    closeModal()
+  }
+
+  const closeModal = () => {
+    onClose()
+  }
+
+  const handleValueChange = (e: React.ChangeEvent<HTMLInputElement>) => setValue(e.target.value)
+  const handleKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Escape") {
+      closeModal()
+    } else if (e.key === "Enter") {
+      applyValue()
+    }
+  }
+  const focusInput = () => inputRef.current?.focus()
+
+  const buttons = [{
+    label: t("DG.AttrFormView.cancelBtnTitle"),
+    tooltip: t("DG.AttrFormView.cancelBtnTooltip"),
+    onClick: closeModal
+  }, {
+    label: t("V3.WebView.Modal.okBtnTitle"),
+    tooltip: t("DG.DocumentController.enterViewWebPageOKTip"),
+    onClick: applyValue
+  }]
+
+  return (
+    <CodapModal
+      closeOnOverlayClick={false}
+      isOpen={isOpen}
+      modalHeight={"140px"}
+      modalWidth={"350px"}
+      onClick={focusInput}
+      onClose={closeModal}
+    >
+      <ModalHeader
+        className="codap-modal-header"
+        data-testid="codap-modal-header"
+        fontSize="md"
+        h="30"
+      >
+        <div className="codap-header-title-simple">
+          {t("DG.DocumentController.enterURLPrompt")}
+        </div>
+      </ModalHeader>
+      <ModalBody>
+        <FormControl display="flex" flexDirection="column" data-testid="web-view-url-form">
+          <Input
+            data-testid="web-view-url-input"
+            onChange={handleValueChange}
+            onKeyDown={e => e.stopPropagation()}
+            onKeyUp={handleKeyUp}
+            placeholder={t("V3.WebView.Modal.PlaceholderText")}
+            ref={r => inputRef.current = r}
```

Comment:
Do you need the functional form here? Seems like the simpler form should work (unless Chakra's forwarding of the ref doesn't work as expected):
```suggestion
            ref={inputRef}
```